formatter.go

```diff
@@ -5,16 +5,26 @@ import (
 	"encoding/json"
 	"fmt"
 	"math"
+	"os"
 	"strconv"
 	"strings"
+	"sync"
+	"testing"
 	"text/template"
 
+	"github.com/fatih/color"
+	"github.com/mattn/go-isatty"
 	"github.com/mitchellh/go-wordwrap"
 	"github.com/sanity-io/litter"
 	"github.com/yudai/gojsondiff"
 	"github.com/yudai/gojsondiff/formatter"
 )
 
+var (
+	once      sync.Once
+	withColor bool
```

Comment:
These vars are global for the whole project, so let's give them some specific names, ideally related to each other, e.g.:

* colorsSupportedGuard
* colorsSupported